examples/lessons/11/app.js

```diff
@@ -0,0 +1,206 @@
+/* eslint-disable no-var, max-statements */
+/* eslint-disable array-bracket-spacing, no-multi-spaces */
+/* global document */
+import {
+  GL, AnimationLoop, loadTextures, addEvents, Vector3, setParameters, Program, Sphere
+} from 'luma.gl';
+
+import {
+  Matrix4
+} from 'math.gl'
+
+const VERTEX_SHADER = `\
+attribute vec3 positions;
+attribute vec3 normals;
+attribute vec2 texCoords;
+
+uniform mat4 uMMatrix;
+uniform mat4 uVMatrix;
+uniform mat4 uPMatrix;
+
+uniform vec3 uAmbientColor;
+
+uniform vec3 uLightingDirection;
+uniform vec3 uDirectionalColor;
+
+uniform bool uUseLighting;
+
+varying vec2 vTextureCoord;
+varying vec3 vLightWeighting;
+
+void main(void) {
+    gl_Position = uPMatrix * uVMatrix * uMMatrix * vec4(positions, 1.0);
+    vTextureCoord = texCoords;
+
+    if (!uUseLighting) {
+        vLightWeighting = vec3(1.0, 1.0, 1.0);
+    } else {
+        vec4 transformedNormal = uMMatrix * vec4(normals, 1.0);
+        vec3 newNormal = transformedNormal.xyz / transformedNormal.w;
+        float directionalLightWeighting = max(dot(newNormal, uLightingDirection), 0.0);
+        vLightWeighting = uAmbientColor + uDirectionalColor * directionalLightWeighting;
+    }
+}
+`;
+
+const FRAGMENT_SHADER = `\
+#ifdef GL_ES
+precision highp float;
+#endif
+
+varying vec2 vTextureCoord;
+varying vec3 vLightWeighting;
+
+uniform sampler2D uSampler;
+
+void main(void) {
+  vec4 textureColor = texture2D(uSampler, vec2(1.0 - vTextureCoord.s, 1.0 - vTextureCoord.t));
+  gl_FragColor = vec4(textureColor.rgb * vLightWeighting, textureColor.a);
+}
+`;
+
+const mouseMovement = {
+  mouseDown: false,
+  lastMouseX: null,
+  lastMouseY: null
+};
+
+const moonRotation = {
+  matrix: new Matrix4()
+};
+
+const animationLoop = new AnimationLoop({
+  onInitialize: ({canvas, gl}) => {
+    addMouseHandler(canvas, mouseMovement, moonRotation);
+
+    setParameters(gl, {
+      clearColor: [0, 0, 0, 1],
+      clearDepth: 1,
+      depthTest: true
+    });
+
+    return loadTextures(gl, {
+      urls: ['moon.gif']
+    })
+    .then(textures => {
+      let program = new Program(gl, {
+        fs: FRAGMENT_SHADER,
```

Comment:
I think you can just add `fs` and `vs` as parameter to `Sphere` and it will create a `Program` internally.

Reply:
cool, good to know that.